py-polars/polars/series/series.py

```diff
@@ -974,8 +974,16 @@ def __getitem__(
 
         # Integer.
         elif isinstance(item, int):
+            if item > self.len() - 1:
+                raise IndexError(
+                    f"Series index {item} is out of bounds (len={self.len()})"
+                )
             if item < 0:
                 item = self.len() + item
+                if item < 0:
+                    raise IndexError(
+                        f"Series index {item - self.len()} is out of bounds (len={self.len()})"
+                    )
```

Comment:
How about a simple
```suggestion
           if (item % self.len) >= self.len:
               raise IndexError(
                  f"Series index {item} is out of bounds (len={self.len()})"
               )
```

Reply:
I see, haven't read the previous replies. I might as well create my own merge request then...